language/move-lang/src/parser/syntax.rs

```diff
@@ -1678,6 +1679,162 @@ fn parse_spec_variable<'input>(tokens: &mut Lexer<'input>) -> Result<SpecBlockMe
+    // parse_comma_list.
+    if tokens.peek() != Tok::NameValue || tokens.content() != "to" {
+        return Err(unexpected_token_error(tokens, "`to`"));
+    }
+    let patterns = parse_comma_list(
+        tokens,
+        Tok::NameValue,
+        Tok::Semicolon,
+        parse_function_pattern,
+        "a function pattern",
+    )?;
+    Ok(spanned(
+        tokens.file_name(),
+        start_loc,
+        tokens.previous_end_loc(),
+        SpecBlockMember_::Apply {
+            name,
+            type_arguments,
+            patterns,
+        },
+    ))
+}
+
+// Parse a function pattern:
+//     FunctionPattern = <NamePatternFragment>+ <OptionalTypeArgs>
+fn parse_function_pattern<'input>(tokens: &mut Lexer<'input>) -> Result<FunctionPattern, Error> {
+    let start_loc = tokens.start_loc();
+    let mut name_pattern = vec![];
+    loop {
+        name_pattern.push(parse_name_pattern_fragment(tokens)?);
+        if ![Tok::NameValue, Tok::Star].contains(&tokens.peek()) {
+            break;
+        }
+    }
+    let type_arguments = parse_optional_type_args(tokens)?;
+    Ok(spanned(
+        tokens.file_name(),
+        start_loc,
+        tokens.previous_end_loc(),
+        FunctionPattern_ {
+            name_pattern,
+            type_arguments,
+        },
+    ))
+}
+
+// Parse a name pattern fragment:
+//     NamePatternFragment = <Name> | "*"
+fn parse_name_pattern_fragment<'input>(
+    tokens: &mut Lexer<'input>,
+) -> Result<NamePatternFragment, Error> {
+    let start_loc = tokens.start_loc();
+    let fragment = match tokens.peek() {
+        Tok::NameValue => NamePatternFragment_::NamePart(parse_name(tokens)?),
+        Tok::Star => {
+            tokens.advance()?;
+            NamePatternFragment_::Wildcard
+        }
+        _ => return Err(unexpected_token_error(tokens, "a name fragment or `*`")),
+    };
+    Ok(spanned(
+        tokens.file_name(),
+        start_loc,
+        tokens.previous_end_loc(),
+        fragment,
+    ))
+}
+
+// Parse a specification pragma:
+//    SpecPragma = "pragma" Comma<SpecPragmaProperty> ";"
+fn parse_spec_pragma<'input>(tokens: &mut Lexer<'input>) -> Result<SpecBlockMember, Error> {
+    let start_loc = tokens.start_loc();
+    // Check for 'pragma' Name value token but don't consume it because we need a start token for
+    // parse_comma_list.
+    if tokens.peek() != Tok::NameValue || tokens.content() != "pragma" {
+        return Err(unexpected_token_error(tokens, "`pragma`"));
+    }
+    let properties = parse_comma_list(
+        tokens,
+        Tok::NameValue,
+        Tok::Semicolon,
+        parse_spec_pragma_property,
+        "a pragma property",
+    )?;
+    Ok(spanned(
+        tokens.file_name(),
+        start_loc,
+        tokens.previous_end_loc(),
+        SpecBlockMember_::Pragma { properties },
+    ))
+}
+
+// Parse a specification pragma property:
+//    SpecPragma = <Name> ( "=" Value )?
```

Comment:
That should be "SpecPragmaProperty = ...."

Reply:
Done.